numpy/lib/npyio.py

```diff
@@ -87,6 +87,13 @@ def __dir__(self):
 
 
 def zipfile_factory(*args, **kwargs):
+    """Create a zipfile.Zipfile. Allows for Zip64, as well as pathlib.Path 
+    objects."""
+    assert len(args) >= 1, "zipfile_factory requires a file"
```

Comment:
I think you can keep the file argument, it isn't optional in the ZipFile constructor and you would get an error anyway without it.
```
class zipfile.ZipFile(file[, mode[, compression[, allowZip64]]])
```